### DISCUSSION TITLE
How do you send a message from a React Popup?

### DISCUSSION THREAD

**Discussion by winstxnhdw** [wxt-dev/wxt#1049] [Q&A]
Looking at the example [here](https://github.com/wxt-dev/examples/blob/main/examples/background-message-forwarder/entrypoints/popup/main.ts), this should work, but it doesn't..

```ts
import './App.css'

export default function App() {
  return (
    <>
      <div className='card'>
        <button type='button' onClick={() => browser.runtime.sendMessage('restore')}>
          Restore
        </button>
      </div>
    </>
  )
}
```

**Comment by 1natsu172** [comment]
Please give me additional information on how it doesn't work.

**Reply by winstxnhdw** [reply]
Ahh, I found out the example never set the `tabs` permission.
